service/src/test/java/bio/terra/cbas/runsets/monitoring/TestSmartRunsPoller.java

```diff
@@ -301,4 +301,54 @@ void updatingOutputFails() throws Exception {
         COMPLETE,
         actual.stream().filter(r -> r.id().equals(completedRunId)).toList().get(0).status());
   }
+
+  @Test
+  void shouldNotCallWDSIfOutputsEmpty() throws Exception {
```

Comment:
TOL, but I wonder whether we could extract some smaller "update WDS" method, so that this test could just target that one piece of functionality, without having to set up a whole complex API test just to check a boolean decision half way through?

Reply:
I can try extracting out the function 👍

Reply:
I extracted the "should it call WDS to update attributes" logic into a separate function that simplified the tests.